staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware_test.go

```diff
@@ -496,9 +498,15 @@ func TestProxyUpgradeErrorResponse(t *testing.T) {
 		expectedErr = errors.New("EXPECTED")
 	)
 	proxy := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		transport := http.DefaultTransport.(*http.Transport)
-		transport.Dial = func(network, addr string) (net.Conn, error) {
```

Comment:
Global mutable state is a bad idea. Mutating it is an even worse idea. Some other test was failing because of that.